safe/gui/tools/people_in_buildings_dialog.py

```diff
@@ -0,0 +1,380 @@
+__revision__ = '$Format:%H$'
+__date__ = '12/02/2015'
+__copyright__ = ('Copyright 2012, Australia Indonesia Facility for '
+                 'Disaster Reduction')
+
+# noinspection PyPackageRequirements
+from PyQt4 import QtGui, QtCore
+from PyQt4.QtCore import QVariant  # , QPyNullVariant
+from qgis.core import QgsField
+from qgis.core import QgsGeometry
+
+from safe.utilities.resources import get_ui_class
+
+FORM_CLASS = get_ui_class('people_in_buildings_base.ui')
+
+People_Calculated = "People_Dec"
+People_Rounded = "People_Rnd"
+
+
+class PeopleInBuildingsDialog(QtGui.QDialog, FORM_CLASS):
+    """Options dialog for the InaSAFE plugin."""
+
+    def __init__(self, iface, parent=None):
+        """Constructor for the dialog.
+
+        :param iface: A Quantum GIS QGisAppInterface instance.
+        :type iface: QGisAppInterface
+
+        :param parent: Parent widget of this dialog
+        :type parent: QWidget
+
+        :param dock: Optional dock widget instance that we can notify of
+            changes to the keywords.
+        :type dock: Dock
+        """
+
+        QtGui.QDialog.__init__(self, parent)
+        self.setupUi(self)
+        # Save reference to the QGIS interface and parent
+        self.iface = iface
+        self.parent = parent
+
+        self.load_layers_into_combo_box()
+
+        self.buildingLayerComboBox.currentIndexChanged['QString'].connect(
+            self.handle_building_layer)
+        self.censusLayerComboBox.currentIndexChanged['QString'].connect(
+            self.handle_census_layer)
+
+        self.buttonBox.button(QtGui.QDialogButtonBox.Apply).clicked.connect(
+            self.estimate_people_in_buildings)
+
+        self.progressBar.setMinimum(0)
+        self.progressBar.setMaximum(100)
+
+    def load_layers_into_combo_box(self):
+        """Load the layer options and column options into the combobox.
+        """
+        layer_names = self._get_layer_names()
+        self._update_combobox(self.buildingLayerComboBox, layer_names)
+        self._update_combobox(self.censusLayerComboBox, layer_names)
+        self.handle_building_layer(self.buildingLayerComboBox.currentText())
+        self.handle_census_layer(self.censusLayerComboBox.currentText())
+
+    def _select_layer_by_name(self, layer_name):
+        """Get a layer by name
+        :param layer_name: The layer's name
+        :type layer_name: basestring
+
+        :returns: The layer that has the appropriate name or none
+        :rtype: QgsMapLayer, None
+        """
+        layers = self._get_layers()
+        layer_names = [layer.name() for layer in layers]
+        if not layer_name in layer_names:
+            return None
+        index = layer_names.index(layer_name)
+        return layers[index]
+
+    def _get_layers(self):
+        """Get all layers.
+
+        :returns: All layers currently loaded.
+        :rtype: list
+        """
+        layers = self.iface.legendInterface().layers()
```

Comment:
I think this will only return visible layers. If you truly want all layers get them from the QgsMapLayerRegistry